### PR TITLE
[NGraph] Add scatterNDUpdate and scatterUpdate reference implementations

### PR DESCRIPTION
**Pull request by mandrono**
Issue: 33266

### CODE REVIEW

**Comment by mandrono** [comment]
@dmitry-gorokhov  @antonvor  @lazarevevgeny  @mvafin take a look please

File: inference-engine/src/mkldnn_plugin/nodes/scatter.cpp [hunk 1]
```diff
@@ -1,188 +0,0 @@
-// Copyright (C) 2018-2020 Intel Corporation
```
Comment (antonvor) [inline inference-engine/src/mkldnn_plugin/nodes/scatter.cpp:1]: Why did you delete "scatter" from plugin?

File: ngraph/test/runtime/interpreter/reference/scatter_nd_update.hpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+//
+
+#pragma once
+
+#include "ngraph/coordinate_transform.hpp"
+#include "ngraph/shape.hpp"
+
+using namespace ngraph;
+
+namespace referenceTest
```
Comment (antonvor) [inline ngraph/test/runtime/interpreter/reference/scatter_nd_update.hpp:12]: Why namespace is "referenceTest"?

File: inference-engine/src/mkldnn_plugin/nodes/scatter.cpp (hunk 1 above)
Comment (antonvor) [reply inference-engine/src/mkldnn_plugin/nodes/scatter.cpp:1]: understood)

File: ngraph/test/runtime/interpreter/reference/scatter_nd_update.hpp (hunk 2 above)
Comment (mandrono) [reply ngraph/test/runtime/interpreter/reference/scatter_nd_update.hpp:12]: It's just namespace for reference implementations that using only for tests and don't used in ngraph backend

Comment (antonvor) [reply ngraph/test/runtime/interpreter/reference/scatter_nd_update.hpp:12]: ok

Comment (ilyachur) [reply ngraph/test/runtime/interpreter/reference/scatter_nd_update.hpp:12]: Can I ask you to add all reference implementations in the namespace as in original nGraph? It should help in future copy past these implementations in the nGraph library.

```
namespace ngraph
{
    namespace runtime
    {
        namespace reference
        {
```

Comment (mandrono) [reply ngraph/test/runtime/interpreter/reference/scatter_nd_update.hpp]: changed namespace

**Comment by mandrono** [comment]
> LGTM, Just a question before the merge. What is the type of MKLDNN primitive used for ScatterNGUpdate? As I can see you removed `ScatterImpl` but you add nothing new.
All types of Scatter primitives used in nGraph implemented here [mkldnn_scatter_update_node.h](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_scatter_update_node.h).
Please don't merge changes until we get approval form @dmitry-gorokhov
